test/structural_natural_bijections.jl

```diff
@@ -0,0 +1,84 @@
+using ChainRulesCore: backing
+
+# Need to be able to compare `Tangent`s to verify correctness of definition of natural.
+Base.:(==)(t::Tangent, s::Tangent) = (backing(t) == backing(s))
+
+# This function specifies what we require of a natural tangent. Works for flat spaces.
+function check_bijections(primal, structural::Tangent)
+    b = Bijections(primal)
+    
+    # Check that adding tangents is the same in structural or natural.
+    sum_1 = to_natural(b, 0.8 * structural + 0.3 * structural)
+    natural = to_natural(b, structural)
+    sum_2 = 0.8 * natural + 0.3 * natural
+    @test sum_1 == sum_2
+
+    # Check that adding the structural tangent to the primal has the same effect as adding
+    # the natural.
+    @test primal + structural == primal + to_natural(b, structural)
+
+    # Check that to_structural inverts to_natural.
+    @test to_structural(b, to_natural(b, structural)) == structural
+end
+
+struct ScaledVector <: AbstractVector{Float64}
+    v::Vector{Float64}
+    α::Float64
+end
+
+Base.getindex(x::ScaledVector, n::Int) = x.α * x.v[n]
```

Comment:
This is an odd type in that it's over-complete. In this case, the map back to the Tangent is not unique, you'd have to choose a gauge slice, I think, and I'd pick `dα = 0`.

Does anything in the wild do that? I thought StepRangeLen might, but it doesn't.

Reply:
> This is an odd type in that it's over-complete. In this case, the map back to the Tangent is not unique, you'd have to choose a gauge slice, I think, and I'd pick dα = 0.

The problem with doing that is that it won't pass the first test in the test suite, because the `α` fields in the `Tangent`s won't generally sum to `0`.

I'm fine with there not existing a natural tangent for this type, as it's all still a mystery to me!

Reply:
There cannot be a bijection here, since the spaces have different dimensions. Unless you constrain to a slice. The to_structural function as written does map into a slice, but whether its choice is kosher or not I'm not super-sure. 

But whether we need to solve this problem at all is the first question. If there are no such types, then we need not care.

Reply:
> There cannot be a bijection here, since the spaces have different dimensions. Unless you constrain to a slice. 

Very good point.

> The to_structural function as written does map into a slice, but whether its choice is kosher or not I'm not super-sure.

Yeah -- the current implementation doesn't pass the three tests, so it probably doesn't make sense.